### ISSUE TITLE
openvpn3 3.8.3 throw 'Main thread exception: OpenSSL: error:1E08003C:DECODER routines::OSSL_DECODER lib' error on macos

### ISSUE THREAD

**Issue by zeddit** [OpenVPN/openvpn3#287]
here is the output.
```
 oathtool -b --totp Y3QBXXXXXXXXXJ5 | sudo ovpncli ~/xxx/xxx.ovpn -u xxx -p $ZPW -x --ssl-debug 7
CONNECTING...
Thread starting...
Sat Nov 25 17:16:45.055 2023 OpenVPN core 3.8.3 mac arm64 64-bit
Sat Nov 25 17:16:45.056 2023 Frame=512/2112/512 mssfix-ctrl=1250
Sat Nov 25 17:16:45.062 2023 EVENT: RESOLVE
Sat Nov 25 17:16:45.089 2023 Contacting xxx:11444 via UDP
Sat Nov 25 17:16:45.089 2023 EVENT: WAIT
NOT IMPLEMENTED: *** socket_protect 7 xxx
Sat Nov 25 17:16:45.089 2023 Connecting to [xxx]:11444 (xxx) via UDP
Sat Nov 25 17:16:45.117 2023 EVENT: CONNECTING
Sat Nov 25 17:16:45.118 2023 SSL state (connect): before SSL initialization
Sat Nov 25 17:16:45.118 2023 SSL state (connect): SSLv3/TLS write client hello
Sat Nov 25 17:16:45.118 2023 Tunnel Options:V4,dev-type tun,link-mtu 1557,tun-mtu 1500,proto UDPv4,cipher AES-256-CBC,auth SHA1,keysize 256,key-method 2,tls-client
Sat Nov 25 17:16:45.118 2023 Creds: Username/Password
Sat Nov 25 17:16:45.118 2023 Sending Peer Info:
IV_VER=3.8.3
IV_PLAT=mac
IV_NCP=2
IV_TCPNL=1
IV_PROTO=990
IV_MTU=1600
IV_CIPHERS=AES-128-GCM:AES-192-GCM:AES-256-GCM:CHACHA20-POLY1305
IV_GUI_VER=cli 1.0
IV_HWADDR=1c:57:dc:49:e0:4a
IV_SSL=OpenSSL 3.2.0 23 Nov 2023

Sat Nov 25 17:16:45.137 2023 SSL state (connect): SSLv3/TLS write client hello
Sat Nov 25 17:16:45.138 2023 SSL state (connect): SSLv3/TLS read server hello
Sat Nov 25 17:16:45.138 2023 SSL state (connect): TLSv1.3 read encrypted extensions
Sat Nov 25 17:16:45.138 2023 SSL state (connect): SSLv3/TLS read server certificate request
Sat Nov 25 17:16:45.138 2023 VERIFY OK: depth=1, /CN=OpenVPN CA, signature: RSA-SHA256
Sat Nov 25 17:16:45.138 2023 VERIFY OK: depth=0, /CN=OpenVPN Server, signature: RSA-SHA256
Sat Nov 25 17:16:45.139 2023 SSL state (connect): SSLv3/TLS read server certificate
Sat Nov 25 17:16:45.139 2023 SSL state (connect): TLSv1.3 read server certificate verify
Sat Nov 25 17:16:45.139 2023 SSL state (connect): SSLv3/TLS read finished
Sat Nov 25 17:16:45.139 2023 SSL state (connect): SSLv3/TLS write change cipher spec
Sat Nov 25 17:16:45.139 2023 SSL state (connect): SSLv3/TLS write client certificate
Sat Nov 25 17:16:45.139 2023 SSL state (connect): SSLv3/TLS write finished
Sat Nov 25 17:16:45.160 2023 SSL state (connect): SSL negotiation finished successfully
Sat Nov 25 17:16:45.160 2023 SSL state (connect): SSL negotiation finished successfully
Sat Nov 25 17:16:45.160 2023 SSL state (connect): SSLv3/TLS read server session ticket
Sat Nov 25 17:16:45.160 2023 SSL state (connect): SSL negotiation finished successfully
Sat Nov 25 17:16:45.160 2023 SSL state (connect): SSL negotiation finished successfully
Sat Nov 25 17:16:45.160 2023 SSL state (connect): SSLv3/TLS read server session ticket
Sat Nov 25 17:16:45.161 2023 SSL Handshake: peer certificate: CN=OpenVPN Server, 2048 bit RSA, cipher: TLS_AES_256_GCM_SHA384         TLSv1.3 Kx=any      Au=any   Enc=AESGCM(256)            Mac=AEAD

Sat Nov 25 17:16:45.161 2023 Session is ACTIVE
Sat Nov 25 17:16:45.161 2023 EVENT: GET_CONFIG
Sat Nov 25 17:16:45.161 2023 Sending PUSH_REQUEST to server...
Sat Nov 25 17:16:46.161 2023 Sending PUSH_REQUEST to server...
Sat Nov 25 17:16:46.179 2023 AUTH_FAILED
Sat Nov 25 17:16:46.179 2023 EVENT: DYNAMIC_CHALLENGE CRV1:R,E:PG_yj+TiqVIwl9WAmG4:emhhbmdkYWk=:Enter Authenticator Code [FATAL-ERR]
DYNAMIC CHALLENGE
challenge: Enter Authenticator Code
echo: 1
responseRequired: 1
stateID: PG_yj+TiqVIwl9WAmG4
Sat Nov 25 17:16:46.179 2023 EVENT: DISCONNECTED
Thread finished
ENTER RESPONSE
Main thread exception: OpenSSL: error:1E08003C:DECODER routines::OSSL_DECODER lib
```

**Comment by zeddit** [comment]
I use openvpn-connect UI to connect to my server instead, and there is no problems. which indicates that my configuration file and username/password is correct.

**Comment by dsommers** [comment]
Which version of OpenVPN Connect do you use?

**Comment by zeddit** [comment]
@dsommers i am using a version of 3.4.0 OpenVPN Connect.
After some diggings, i found it's the problem of the underlying openssl lib.
openvpn3.8.3 cannot work well with openssl 3.2.0, when changing the openssl lib to v1.1, the problem disappears.

**Comment by schwabe** [comment]
Yes. And the comment you get has the same overall sentiment as mine: https://trac.macports.org/ticket/68776#comment:4 to not having enough information.

@zeddit errors like these can very well be certificates or other settings where OpenSSL 3 is stricter. But unless you provide us with more information, there is nothing we can do.

**Comment by zeddit** [comment]
@schwabe I agree with you that I should make the system reproducible.

However, I don't have much details about the configurations.
What I know is that the server is openvpn access server and it enables two factor auth, and I download the ovpn configuration file directly from the openvpn_as web page after login.

could you advice me some steps to get more information to debug the system. great thanks.

**Comment by novaflash** [comment]
The best thing you can do then is contact OpenVPN Inc. who created and supports OpenVPN Access Server and provide all the relevant details there. They can also help you to gather the necessary information and send it to them securely instead of publicly posted on github. You can reach their support at https://support.openvpn.net/

I am sure they would be happy to take a look at try to figure out what's going on there. OpenVPN Connect v3 and OpenVPN Access Server are supposed to work correctly together, they're created and supported by the same company.

**Comment by dvalter** [comment]
I had the same issue with openvpn2 on Linux. It is caused by https://github.com/openssl/openssl/issues/22939 and the proposed fix solves it for me.

**Comment by zeddit** [comment]
@dvalter the issue is fixed after I have upgraded openssl3 to 3.2.0_1 and openvpn3 to 3.8.3_5.
great thanks.
